File: apps/scraper/query/sources/jobs.py
"""On-campus student employment listings."""

from __future__ import annotations

from scraper.core.types import QuerySource
from scraper.sources.jobs import SOURCE as JOBS


def to_url(params: dict[str, str]) -> str:
    """The student employment page. It takes no parameters."""
    return JOBS.url


QUERY = QuerySource(
    key="jobs",
    description="Fetch ASU student employment listings and how to apply.",
    params=(),
    to_url=to_url,
    needs_js=JOBS.needs_js,
    extractor=JOBS.extractor,
)

File: apps/engine/src/runtime/tools/knowledge/search/jobs.rs
//! ASU student employment listings and how to apply.

use super::{LiveSource, Param};

/// The ASU student employment page.
pub struct Jobs;

impl LiveSource for Jobs {
    fn key(&self) -> &'static str {
        "jobs"
    }

    fn hint(&self) -> &'static str {
        "student employment listings"
    }

    fn label(&self) -> &'static str {
        "ASU Student Employment"
    }

    fn category(&self) -> &'static str {
        "jobs"
    }

    fn params(&self) -> &'static [Param] {
        &[]
    }
}
